examples/fabric-admin/device_manager/DeviceSubscriptionManager.cpp

```diff
@@ -34,22 +34,53 @@ namespace {
 
 void OnDeviceConnectedWrapper(void * context, Messaging::ExchangeManager & exchangeMgr, const SessionHandle & sessionHandle)
 {
-    reinterpret_cast<DeviceSubscription *>(context)->OnDeviceConnected(exchangeMgr, sessionHandle);
+    reinterpret_cast<DeviceSubscriptionManager::DeviceSubscription *>(context)->OnDeviceConnected(exchangeMgr, sessionHandle);
 }
 
 void OnDeviceConnectionFailureWrapper(void * context, const ScopedNodeId & peerId, CHIP_ERROR error)
 {
-    reinterpret_cast<DeviceSubscription *>(context)->OnDeviceConnectionFailure(peerId, error);
+    reinterpret_cast<DeviceSubscriptionManager::DeviceSubscription *>(context)->OnDeviceConnectionFailure(peerId, error);
 }
 
 } // namespace
 
-DeviceSubscription::DeviceSubscription() :
+DeviceSubscriptionManager & DeviceSubscriptionManager::Instance()
+{
+    static DeviceSubscriptionManager instance;
+    return instance;
+}
+
+CHIP_ERROR DeviceSubscriptionManager::StartSubscription(Controller::DeviceController & controller, NodeId nodeId)
+{
```

Comment:
std::lock_guard<std::mutex> lock(mMapMutex);
        // ... (existing code)

Reply:
Use assertChipStackLockedByCurrentThread also works for me if we don't want to use extra guard

Reply:
Discussed offline `assertChipStackLockedByCurrentThread` is preferred since we are already called by thread that has a lock. Also added comment explicitly stating that mDeviceSubscriptionMap should only be accessed by something that holds the ChipStackLock